<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <ros2_control name="GazeboSystem" type="system">
        <hardware>
            <plugin>gazebo_ros2_control/GazeboSystem</plugin>
        </hardware>
        <joint name="joint_zx">
            <command_interface name="position">
                <param name="min">-3</param>
                <param name="max">3</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="joint_xy">
            <command_interface name="position">
                <param name="min">-3</param>
                <param name="max">3</param>
            </command_interface>
            <state_interface name="position"/> <!--what i want to broadcast-->
            <state_interface name="velocity"/>
        </joint>
        <joint name="joint_needle">
            <command_interface name="position">
                <param name="min">-3</param>
                <param name="max">3</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="joint_2_x">
            <command_interface name="position">
                <param name="min">-3</param>
                <param name="max">3</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="joint_2_z">
            <command_interface name="position">
                <param name="min">-3</param>
                <param name="max">3</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="joint_3_x">
            <command_interface name="position">
                <param name="min">-3</param>
                <param name="max">3</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="joint_3_z">
            <command_interface name="position">
                <param name="min">-3</param>
                <param name="max">3</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="joint_4_x">
            <command_interface name="position">
                <param name="min">-3</param>
                <param name="max">3</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="joint_4_z">
            <command_interface name="position">
                <param name="min">-3</param>
                <param name="max">3</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="joint_5_x">
            <command_interface name="position">
                <param name="min">-3</param>
                <param name="max">3</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="joint_5_z">
            <command_interface name="position">
                <param name="min">-3</param>
                <param name="max">3</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
    </ros2_control>

    <gazebo>
        <plugin filename="libgazebo_ros2_control.so" name="gazebo_ros2_control">
            <parameters>$(find my_bot)/config/controller.yaml</parameters>
        </plugin>
    </gazebo>



</robot>
